src/native.rs

```diff
@@ -1414,6 +1415,164 @@ impl super::Context for Context {
             pass as u32,
         );
     }
+
+    unsafe fn debug_message_control(
+        &self,
+        source: u32,
+        msg_type: u32,
+        severity: u32,
+        ids: Option<&[u32]>,
+        enabled: bool,
+    ) {
+        let gl = &self.raw;
+        let (count, ids_ptr) = ids
+            .map(|i| (i.len(), i.as_ptr()))
+            .unwrap_or((0, std::ptr::null()));
+        gl.DebugMessageControl(
+            source,
+            msg_type,
+            severity,
+            count as i32,
+            ids_ptr,
+            enabled as u8,
+        );
+    }
+
+    unsafe fn debug_message_insert<S>(
+        &self,
+        source: u32,
+        msg_type: u32,
+        id: u32,
+        severity: u32,
+        msg: S,
+    ) where
+        S: AsRef<str>
+    {
+        let gl = &self.raw;
+        let message = CString::new(msg.as_ref()).unwrap();
```

Comment:
nit: I don't think conversion to `CString` is necessary here. It involves heap allocation and aims to provide a null-terminated string. This method is documented to work with both terminated and non-terminated strings:
> If <length> is negative, it is implied that <buf> contains a null terminated string.

It may be important to keep these methods light, since there could be scenarios where this debug output is enabled in optimized builds.